conn.go

```diff
@@ -144,35 +148,87 @@ type DialContext struct {
 	tc *tls.Config
 }
 
-func (dc *DialContext) dial(u *url.URL) (net.Conn, error) {
+func (dc *DialContext) dial(u *url.URL) (conn net.Conn, err error) {
```

Comment:
May you update the related tests and add some comments for documentation. The newly added functionality isn't really self-explanatory, especially when it comes down to the `rootDN` and `baseDN`